BundleClient/app/build.gradle

```diff
@@ -1,85 +1,107 @@
-apply plugin: 'com.android.application'
-apply plugin: 'com.google.protobuf'
+plugins {
+    id 'com.android.application'
+    id "com.google.protobuf" version '0.9.0'
+}
 
 android {
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
-    compileSdkVersion 29
+
+    compileSdk 31
 
     defaultConfig {
         applicationId "com.ddd.bundleclient"
-        minSdkVersion 24
-        targetSdkVersion 29
+        minSdk 24
+        targetSdk 31
         versionCode 1
         versionName "1.0"
     }
+
     buildTypes {
         debug { minifyEnabled false }
         release {
-            minifyEnabled true
+            minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+
     lintOptions {
         abortOnError false
         disable 'GoogleAppIndexingWarning', 'HardcodedText', 'InvalidPackage'
         textReport true
         textOutput "stdout"
     }
-}
 
-protobuf {
-    protoc { artifact = 'com.google.protobuf:protoc:3.17.3' }
-    plugins {
-        grpc { artifact = 'io.grpc:protoc-gen-grpc-java:1.50.2' // CURRENT_GRPC_VERSION need to change for windows, MAC specific
-        }
-    }
-    generateProtoTasks {
-        all().each { task ->
-            task.builtins {
-                java { option 'lite' }
+    sourceSets {
+        main {
+            proto {
+                // srcDir 'src/main/proto'
             }
-            task.plugins {
-                grpc { // Options added to --grpc_out
-                    option 'lite' }
+            java {
+                // srcDir 'src/main/java'
             }
         }
     }
+    namespace 'com.ddd.bundleclient'
 }
 
 dependencies {
+    implementation 'io.grpc:grpc-netty-shaded:1.50.2'
+    implementation 'io.grpc:grpc-protobuf:1.50.2'
+    implementation 'io.grpc:grpc-stub:1.50.2'
+    compileOnly 'org.apache.tomcat:annotations-api:6.0.53' // necessary for Java 9+
```

Comment:
do we actually need netty and tomcat? (perhaps okhttp needs netty?)

Reply:
We need Tomcat but not Netty. Removed netty in the latest commit.